tests/test_custom_classes.py

```diff
@@ -110,3 +113,29 @@ class CustomGroup(click.Group):
     group = CustomGroup()
     subgroup = group.group()(lambda: None)
     assert type(subgroup) is CustomGroup
+
+
+def test_custom_exception_class():
+    """Subclasses of click.ClickException should be able to override
+    the default message start "Error: {message}" """
+
+    class CustomException(click.ClickException):
+        def format_message(self) -> str:
+            return f"CustomError: {self.message}"
+
+    class MockStdIO(IO, metaclass=ABCMeta):
+        value = None
+
+        def write(self, *args, **kwargs):
+            self.value = args[0]
+
+    try:
+        raise CustomException("Bad things happened")
```

Comment:
Don't need to raise, can call `CustomException("...").show(f)`

Reply:
Alright. I'll resolve